util/src/main/java/io/kubernetes/client/util/WebSockets.java

```diff
@@ -0,0 +1,107 @@
+package io.kubernetes.client.util;
+
+import io.kubernetes.client.ApiClient;
+import io.kubernetes.client.ApiException;
+import io.kubernetes.client.Pair;
+
+import com.squareup.okhttp.MediaType;
+import com.squareup.okhttp.Request;
+import com.squareup.okhttp.Response;
+import com.squareup.okhttp.ResponseBody;
+import com.squareup.okhttp.ws.WebSocket;
+import com.squareup.okhttp.ws.WebSocketListener;
+
+import static com.squareup.okhttp.ws.WebSocket.BINARY;
+import static com.squareup.okhttp.ws.WebSocket.TEXT;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.Reader;
+import java.util.ArrayList;
+import java.util.HashMap;
+
+import okio.Buffer;
+
+public class WebSockets {
+    /**
+     * A simple interface for a listener on a web socket
+     */
+    public interface SocketListener {
+        /**
+         * Called when the socket is opened
+         */
+        public void open();
+
+        /**
+         * Callled when a binary media type message is received
+         * @param in The input stream containing the binary data
+         */
+        public void bytesMessage(InputStream in);
+
+        /**
+         * Called when a text media type message is received
+         * @param in The character stream containing the message
+         */
+        public void textMessage(Reader in);
+
+        /**
+         * Called when the stream is closed.
+         */
+        public void close();
+    }
+
+    /**
+     * Create a new WebSocket stream
+     * @param path The HTTP Path to request from the API
+     * @param method The HTTP method to use for the call
+     * @param client The ApiClient for communicating with the API
+     * @param listener The socket listener to handle socket events
+     */
+    public static void stream(String path, String method, ApiClient client, SocketListener listener) throws ApiException, IOException {
+        HashMap<String, String> headers = new HashMap<String, String>();
+        headers.put("X-Stream-Protocol-Version", "v4.channel.k8s.io,v3.channel.k8s.io,v2.channel.k8s.io,channel.k8s.io");
+        headers.put("Connection", "Upgrade");
```

Comment:
Do you think it is cleaner to have all string literals as constant? If you agree, we can make it an style guide. Of course there could be exceptions (that should be well defined).